docs/plugins/repository-s3.asciidoc

```diff
@@ -145,6 +145,18 @@ settings belong in the `elasticsearch.yml` file.
     Whether retries should be throttled (i.e. should back off). Must be `true`
     or `false`. Defaults to `true`.
 
+`path_style_access`::
+
+   Whether to use the path style access pattern. If `true`, the path style access pattern will be used if false, DNS style access will
+   be used. Defaults to letting the AWS S3 SDK decide the right access pattern dynamically.
+
+NOTE: This setting is deprecated and only intended as a stop-gap solution to provide
+   compatibility with alternative S3 implementations that do not provide compatibility with the domain style access pattern.
+   AWS S3 will stop supporting the path style access pattern for new buckets from
+   https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story[September 30th, 2020].
+   Any setups using the path style access pattern should be switched to the domain style access pattern as soon as possible to ensure
+   continued compatibility with the S3 repository plugin.
```

Comment:
Isn't toning it down a little inconsistent with us now making a breaking change in `7.x`?
My thinking was that we're doing this breaking change because we fear that future versions of the SDK won't support path style access. If we don't think this is an imminent enough issue to strongly push users towards adjusting their environment asap then why make a breaking change and not just wait for `8.0`?

Reply:
> Isn't toning it down a little inconsistent with us now making a breaking change in 7.x?

It's just going back to the pre-7.0 behavior, which was the more sensible default, especially given the current strategy of Amazon not to support path-style access in the future anymore, and there being no way to explicitly force-disable it in the SDK (i.e. it would still use path-style access with IP-based endpoint). This here makes it sound like we would break Minio support with path-style access at any moment in time, which sounds perhaps a bit overdramatic.